### ISSUE TITLE
Video format VP9 is not supported

### ISSUE THREAD

**Issue by Soberia** [staxrip/staxrip#85]
Crash when open files with VP9 codec (FFVideoSource)

[StaxRip Log]:
```
-------------------------- System Environment --------------------------

StaxRip    : 2.0.6.0
Windows    : Windows 10 Pro 1903
Language   : English (United States)
CPU        : Intel(R) Core(TM) i7-3632QM CPU @ 2.20GHz
GPU        : NVIDIA GeForce GT 635M, Intel(R) HD Graphics 4000
Resolution : 1366 x 768
DPI        : 96

------------------------ MediaInfo Source File ------------------------

D:\1.mkv

General
Complete name        : D:\1.mkv
Format               : Matroska
Format version       : Version 4
File size            : 68.6 MiB
Duration             : 20 min 35 s
Overall bit rate     : 466 kb/s
Encoded date         : UTC 2019-11-09 23:59:40
Writing application  : mkvmerge v40.0.0 ('Old Town Road + Pony') 64-bit
Writing library      : libebml v1.3.9 + libmatroska v1.5.2

Video
ID                   : 1
Format               : VP9
Codec ID             : V_VP9
Duration             : 20 min 35 s
Bit rate             : 340 kb/s
Width                : 640 pixels
Height               : 360 pixels
Display aspect ratio : 16:9
Frame rate mode      : Constant
Frame rate           : 25.000 FPS
Bits/(Pixel*Frame)   : 0.059
Stream size          : 50.0 MiB (73%)
Default              : Yes
Forced               : No

Audio
ID                   : 2
Format               : Opus
Codec ID             : A_OPUS
Duration             : 20 min 35 s
Bit rate             : 124 kb/s
Channel(s)           : 2 channels
Channel layout       : L R
Sampling rate        : 48.0 kHz
Frame rate           : 50.000 FPS (960 SPF)
Compression mode     : Lossy
Stream size          : 18.2 MiB (27%)
Language             : English
Default              : Yes
Forced               : No

------------------ Demux MKV using mkvextract 39.0.0 ------------------

"C:\Program Files\StaxRip\Apps\Support\MKVToolNix\mkvextract.exe" D:\1.mkv tracks 1:"D:\1_temp\1 ID1 English.opus" --ui-language en

Extracting track 1 with the CodecID 'A_OPUS' to the file 'D:\1_temp\1 ID1 English.opus'. Container format: Ogg (Opus in Ogg)

Start:    3:30:36 AM
End:      3:30:36 AM
Duration: 00:00:00

General
Complete name    : D:\1_temp\1 ID1 English.opus
Format           : Ogg
File size        : 18.5 MiB
Duration         : 20 min 35 s
Overall bit rate : 125 kb/s

Audio
ID               : 1686078688 (0x647F84E0)
Format           : Opus
Duration         : 20 min 35 s
Channel(s)       : 2 channels
Channel layout   : L R
Sampling rate    : 48.0 kHz
Compression mode : Lossy
Writing library  : unknown encoder; extracted from Matroska with mkvextract v40.0.0 ('Old Town Road + Pony') 64-bit

----------------------- Indexing using ffmsindex -----------------------

"C:\Program Files\StaxRip\Apps\Plugins\Both\FFMS2\ffmsindex.exe" D:\1.mkv D:\1_temp\1.ffindex

Writing index... done.

Start:    3:30:36 AM
End:      3:30:37 AM
Duration: 00:00:00

------------------------------ Exception ------------------------------

System.NotImplementedException: Video format VP9 is not supported.
   at StaxRip.VideoStream.get_Ext() in D:\Projekte\VB\staxrip\General\Misc.vb:line 1740
   at StaxRip.VideoStream.get_ExtFull() in D:\Projekte\VB\staxrip\General\Misc.vb:line 1745
   at StaxRip.NullEncoder.get_OutputPath() in D:\Projekte\VB\staxrip\Encoding\VideoEncoder.vb:line 720
   at StaxRip.NullEncoder.get_OutputExt() in D:\Projekte\VB\staxrip\Encoding\VideoEncoder.vb:line 730
   at StaxRip.MainForm.Assistant() in D:\Projekte\VB\staxrip\Forms\MainForm.vb:line 2699
   at StaxRip.MainForm.OpenVideoSourceFiles(IEnumerable`1 files, Boolean isEncoding) in D:\Projekte\VB\staxrip\Forms\MainForm.vb:line 2177
```

[AVISynth Log]:
```
LoadPlugin("C:\Program Files\StaxRip\Apps\Plugins\Both\FFMS2\ffms2.dll")
FFVideoSource("D:\1.mkv", colorspace = "YV12", \
              cachefile = "D:\1_temp\1.ffindex")
ConvertToYV12()
```

**Comment by Patman86** [comment]
Hi Soberia,

please test these versions
[ffms_aom](http://forum.doom9.org/showthread.php?p=1883905#post1883905) or [ffms_dav1d](http://forum.doom9.org/showthread.php?p=1886890#post1886890)

**Comment by Soberia** [comment]
Unfortunately not for me
Here's a [sample file](http://www.mediafire.com/file/ywfe4ofpspgthac/Sonic_The_Hedgehog.mkv/file) if you want to give a try
Thanks btw

**Comment by Patman86** [comment]
Thanks for the sample file. Indexing your sample with ffms works great for me. Do you use avisynth filters?

**Comment by Soberia** [comment]
I tried with both of AviSynth and VapourSynth
Strange behavior! Maybe I do something wrong.
![Annotation 2019-11-13 213443](https://user-images.githubusercontent.com/30440494/68791228-1e1b0080-065e-11ea-98ed-5ee239b0730c.png)

**Comment by Patman86** [comment]
Thanks for your screenshot. I found the bug. Copy / mux a video file with vp9 codec into mp4 is currently not supported by staxrip. MP4box can handle it. Adjustments in the Staxrip code are required.

**Comment by stax76** [comment]
Hello, it will be fixed in 2.0.7.4
